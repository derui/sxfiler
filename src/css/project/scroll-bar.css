.scroll-bar__root {
  position: absolute;
  right: 0;
  bottom: 0;
  top: 0;
  width: var(--space-small);
  height: 100%;
  background-color: transparent;
  padding: var(--space-base);
}

.scroll-bar__bar {
  position: relative;
  display: inline-block;
  border-radius: var(--space-small);
  /* background-color: rgb(var(--color-base01)); */
  background-color: var(--color-scrollBar-background);
}
