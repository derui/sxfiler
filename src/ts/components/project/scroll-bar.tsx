import { h } from "preact";

export type Props = {
  start: number;
  windowSize: number;
};

export const Component: preact.FunctionComponent<Props> = ({ start, windowSize }) => {
  const top = start * 100.0;
  const height = windowSize * 100.0;

  return (
    <div class="scroll-bar__root" data-testid="scrollBar-root">
      <div
        class="scroll-bar__bar"
        data-testid="scrollBar-bar"
        style={{
          top: `${top}%`,
          height: `${height}%`,
        }}
      />
    </div>
  );
};
