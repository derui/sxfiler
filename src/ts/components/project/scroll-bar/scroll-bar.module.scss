@import "../../ui/commom-style/_colors.scss";
@import "../../ui/commom-style/_variables.scss";
@import "../../ui/commom-style/reset.module.css";

.root {
  position: absolute;
  right: 0;
  bottom: 0;
  top: 0;

  width: $smallSpace;
  height: 100%;

  background-color: transparent;

  padding: $baseSpace;
}

.bar {
  position: relative;
  display: inline-block;

  border-radius: $smallSpace;
  background-color: $sol_base01;
}
