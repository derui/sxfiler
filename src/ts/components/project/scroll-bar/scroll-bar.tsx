import * as React from "react";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require("./scroll-bar.module.scss");

export interface Props {
  start: number;
  windowSize: number;
}

export const Component: React.FC<Props> = ({ start, windowSize }) => {
  const top = `${start * 100.0}%`;
  const height = `${windowSize * 100.0}%`;

  return (
    <div className={styles.root}>
      <div className={styles.bar} style={{ top, height }} />
    </div>
  );
};
