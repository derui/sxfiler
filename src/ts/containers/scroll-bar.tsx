import * as React from "react";

interface Prop {
  start: number;
  windowSize: number;
}

const ScrollBar : React.FC<Prop> = ({start, windowSize}) => {
  const top = `${start * 100.0}%`;
  const height = `${windowSize * 100.0}%`;

  return (
    <div className="fp-ScrollBar">
      <span className="fp-ScrollBar_VisibleWindow"/>
    </div>
  );
};

export default ScrollBar;
