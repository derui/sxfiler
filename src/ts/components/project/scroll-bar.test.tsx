import { h } from "preact";
import { render } from "preact-render-to-string";
import { Component } from "./scroll-bar";

describe("Project", () => {
  describe("Scroll Bar", () => {
    test("render for window", () => {
      const component = render(<Component start={1} windowSize={0.7} />);

      expect(component).toMatchSnapshot();
    });
  });
});
