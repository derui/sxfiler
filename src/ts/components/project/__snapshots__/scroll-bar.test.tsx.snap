// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Scroll Bar render for window 1`] = `"<div class=\\"scroll-bar__root\\" data-testid=\\"scrollBar-root\\"><div class=\\"scroll-bar__bar\\" data-testid=\\"scrollBar-bar\\" style=\\"top: 100%; height: 70%;\\"></div></div>"`;
